main.tf

```diff
@@ -28,19 +32,15 @@ resource "aws_subnet" "public" {
     create_before_destroy = true
   }
 
-  tags {
-    Name = "${var.name}.${element(var.azs, count.index)}"
-  }
+  tags = "${merge(var.tags, map("Name", format("%s-subnet-public-%s", var.name, element(var.azs, count.index))))}"
 }
 
 # Routes
 resource "aws_route_table" "public" {
   vpc_id = "${var.vpc_id}"
   count  = "${length(var.cidrs)}"
 
-  tags {
-    Name = "${var.name}.${element(var.azs, count.index)}"
-  }
+  tags = "${merge(var.tags, map("Name", format("%s-rt-public-%s", var.name, element(var.azs, count.index))))}"
```

Comment:
Naming of things is a hard one, you know :) I see your point and I like short names, because all resources are generally being identified by combination of `type` and `name` (or `id`) (this case `name` already include `subnet`, for example).

Thanks for your PRs!